lib/tortoise/connection.ex

```diff
@@ -568,6 +586,32 @@ defmodule Tortoise.Connection do
     %State{state | keep_alive: nil}
   end
 
+  defp update_last_will(%State{connect: connect, handler: handler} = state) do
+    case last_will_from_handler(handler) do
+      {last_will, _updated_handler} when last_will != nil ->
+        updated_connect = %Connect{connect | will: last_will}
+
+        %State{state | connect: updated_connect}
+
+      _other ->
```

Comment:
I would prefer if the `_other` catch-all match on cases we expect, such that it otherwise would explode and we will hear about the unexpected cases. Also, perhaps we could consider inlining the `last_will_from_handler/1` function; I don't know if the helper function improves readability.

Reply:
Fixed